### PR TITLE
Implement unmark order command

### LINKED ISSUE #96: Implement unmark order command
To be done after `mark` command is implemented and merged.
Command format: `unmark ORDER_INDEX`

### PR DESCRIPTION
**Pull request by yu-sutong**
close #96

### CODE REVIEW

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Person selectedPerson = model.getSelectedPerson().get();
+        if (selectedPerson == null) {
+            throw new CommandException(MESSAGE_ORDERLIST_DOES_NOT_EXIST);
+        }
+
+        OrderList orderList = selectedPerson.getOrderList();
```
Comment (FooChao) [inline src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java:59]: it shouldn't be the class Order List. size but the filtered list . size ( I told u if u pr after kx u have to fix it yourself)

File: src/test/java/seedu/sellsavvy/logic/parser/UnmarkOrderCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import seedu.sellsavvy.logic.commands.ordercommands.UnmarkOrderCommand;
+
+/**
+ * As we are only doing white-box testing, our test cases do not cover path variations
+ * outside of the UnmarkOrderCommand code. For example, inputs "1" and "1 abc" take the
+ * same path through the DeletePersonCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
+
```
Comment (FooChao) [inline src/test/java/seedu/sellsavvy/logic/parser/UnmarkOrderCommandParserTest.java:19]: why extra space here

File: docs/UserGuide.md [hunk 3]
```diff
@@ -206,6 +206,21 @@ Examples:
 * `markOrder 1` marks the order with index 1 from the displayed order list as completed.
 * `listOrder 1` followed by `markOrder 2` selects the 1st person in the address book and marks the 2nd order under the 1st person as completed.
 
+### Reverts an order to pending status : `unmarkOrder`
+
+Reverts an order under the person whose orders are displayed as pending.
+
+Format: `unmarkOrder ORDER_INDEX`
+
+* A person's order list must be selected using `listOrder` before marking an order from that person.
```
Comment (FooChao) [inline docs/UserGuide.md:215]: image
here is some phrasing that kx changed which i think makes sense with his new command

Comment (FooChao) [reply docs/UserGuide.md:215]: do change to standardise

File: docs/UserGuide.md [hunk 4]
```diff
@@ -267,4 +282,5 @@ Action                | Format, Examples
 **List all orders**             | `listOrder INDEX`<br> e.g., `listOrder 3`
 **Delete an order**             | `deleteOrder ORDER_INDEX`<br> e.g., `deleteOrder 2`
 **Mark an order as completed**  | `markOrder ORDER_INDEX`<br> e.g., `markOrder 2`
+**Unmark an order as completed**| `unmarkOrder ORDER_INDEX`<br> e.g., `unmarkOrder 2`
```
Comment (kaixuan477) [inline docs/UserGuide.md:285]: As per unmarkCommand messages, unmark should be understood as removing the `Completed` status/mark

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            "The order has been reverted to the pending status: %1$s";
+    public static final String MESSAGE_ORDER_ALREADY_UNMARKED_WARNING = "Note: "
+            + "This order is already in the pending status"
+            + ", verify if this is a mistake\n";
+
+    private final Index index;
+
+    /**
+     * Creates a UnmarkOrderCommand to revert the order of the
+     * specified index to the pending status.
```
Comment (kaixuan477) [inline src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java:42]: Not sure why I can't make suggestion on 2 lines but change to:
```
Creates a UnmarkOrderCommand to revert the status of an order at the specified index to pending.
```

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java (hunk 1 above)
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java:59]: Also in line 54, likely you have to refactor that as well so just take note.

**Review by kaixuan477 — COMMENTED** [review]
Looks fine, just minor things to note.

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+        OrderList orderList = selectedPerson.getOrderList();
+        if (index.getZeroBased() >= orderList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+        }
+
+        Order orderToUnmark = orderList.get(index.getZeroBased());
+        Order newOrder = createUnmarkedOrder(orderToUnmark);
+        orderList.setOrder(orderToUnmark, newOrder);
+        String feedbackToUser = (orderToUnmark.getStatus() == Status.PENDING)
```
[the remark this answers is not in the document]
Comment (awdse22) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java:67]: As per discussion, will resolve this. However, do note that the implementation of MarkOrderCommand should be changed to match this as well, are you able to help with that?

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    @Test
+    public void execute_noOrderListDisplayed_throwsCommandException() {
+        UnmarkOrderCommand unmarkOrderCommand = new UnmarkOrderCommand(INDEX_FIRST_ORDER);
+        model.updateSelectedPerson(null);
+
+        assertCommandFailure(unmarkOrderCommand, model, Messages.MESSAGE_ORDERLIST_DOES_NOT_EXIST);
+    }
+
+    @Test
+    public void execute_orderAlreadyUnmarked_warningGiven() {
```
[the remark this answers is not in the document]
Comment (awdse22) [reply src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java:71]: will resolve this as per discussion.

[the remark this answers is not in the document]
Comment (awdse22) [reply src/test/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommandTest.java:71]: This is fine as per discussion, but remember to update test case for MarkOrderCommand as well.

**Review by awdse22 — APPROVED** [review]

**Pull request merged**